### PR TITLE
Repeater to resend case by external ID

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

This management command is needed by a project who fixed a configuration problem in DHIS2, and needs to resend payloads for a bunch of cases. The cases need to be identified by their DHIS2 IDs.

Context: [Jira](https://dimagi-dev.atlassian.net/browse/SC-2330)

## Feature Flag

While the issue motivating this PR is for a DHIS2 integration, the script would work for any data forwarding. Data forwarding requires Pro Plan or higher.

### Safety story

* Management command
* Custom code
* Tested manually

### Automated test coverage

The management command is not covered by automated tests, but pretty much everything it does is.

### QA Plan

No QA planned

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Charl1996 — APPROVED** [review]
No issues from my side. 

Although one question (non-blocking; may be not even the right platform to ask this): what is the external_id of a case? Is it the id of a specific case on an external system?

**Review by sravfeyn — APPROVED** [review]

File: custom/hmhb/management/commands/repeat_forms_by_external_id.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        repeater = Repeater.get(options['repeater_id'])
+        assert repeater.domain == options['domain'], 'Repeater not found'
+
+        case = CommCareCase.objects.get_case_by_external_id(
+            options['domain'],
+            options['external_id'],
+            options['case_type'],
+            raise_multiple=True,
+        )
+        assert case, f'Case {options["external_id"]!r} not found'
```
Comment (kaapstorm) [inline custom/hmhb/management/commands/repeat_forms_by_external_id.py:47]: Stickler is failing here with "E999 SyntaxError: invalid syntax"

I don't see a syntax error here (and this line works both with and without an AssertionError). What am I missing?

**Comment by kaapstorm** [comment]
@Charl1996 
> what is the external_id of a case? Is it the id of a specific case on an external system?

Yes, that's exactly what it's used for. The important thing about external_id is that it's indexed, and so it's efficient when using it to retrieve cases. (Also useful to keep in mind: uniqueness is _not_ enforced.)

File: custom/hmhb/management/commands/repeat_forms_by_external_id.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            'a repeater for each form that updated that case.')
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('repeater_id')
+        parser.add_argument('external_id')
+        parser.add_argument('case_type')
+
+    def handle(self, *args, **options):
+        repeater = Repeater.get(options['repeater_id'])
```
Comment (millerdev) [inline custom/hmhb/management/commands/repeat_forms_by_external_id.py:38]: Should this use `SQLRepeater` when https://github.com/dimagi/commcare-hq/pull/32160 is merged? (I think that will happen soon.)

Comment (kaapstorm) [reply custom/hmhb/management/commands/repeat_forms_by_external_id.py:38]: Yes. Thank you for flagging.

**Pull request merged**